Source/VolunteerReporting/Read/CaseReportsForListing/CaseReportsForListing.cs

```diff
@@ -26,6 +33,26 @@ public CaseReportsForListing(IMongoDatabase database)
             return await list.ToListAsync();
         }
 
+        public async Task<IEnumerable<CaseReportForListing>> GetLimitAsync(int limit, Boolean last)
+        {
+            
+            var filter = Builders<CaseReportForListing>.Filter.Empty;
+
+            // TODO: Can we assume that this explicit conversion is safe?
+            var totalDocuments = (int)await _collection.CountAsync(filter);
+
+            limit = (limit > totalDocuments) ? totalDocuments : limit;
+
+            var options = new FindOptions<CaseReportForListing>()
+            {
+                Skip = last ? totalDocuments - limit : 0,
```

Comment:
Is this correct? If you don't want the first 100, but the following 100 in a collection of 1000, you enter 100 and last = true. That means 1000 - 100 = 900, i.e. you skip the first 900, not just 100. If you enter 900, then you skip the first 100, but it sounds strange with a limit of 900 to get 100. Maybe limit is not right word? Am I missing something?

Reply:
Well, limit is just the max amount of reports you want to fetch from the collection. And depending on what you want you can retrieve the limit last elements or the limit first

Reply:
In my implementation here there is not really room for getting the "in between" documents...
It all very much depends on what you want the system to do really.

Reply:
Limit= 900 and last = true would fetch the last 900 documents, skipping the first Collection.Size - 900 documents.